packages/firestore/src/remote/remote_store.ts

```diff
@@ -340,18 +282,24 @@ export class RemoteStore {
       onClose: this.onWatchStreamClose.bind(this),
       onWatchChange: this.onWatchStreamChange.bind(this)
     });
+
+    this.onlineStateTracker.handleWatchStreamStart();
   }
 
   /**
-   * Returns whether the watch stream should be started because there are
-   * active targets trying to be listened too
+   * Returns whether a watch stream is currently necessary (network is enabled
+   * and we have active watch targets).
+   */
+  private isWatchStreamNecessary(): boolean {
```

Comment:
I'm not a fan of this extra concept that's confusingly similar to shouldStartWatchStream, especially since it's only used in the implementation of it.